lambda/update/attorney_opt_out.go

```diff
@@ -0,0 +1,45 @@
+package main
+
+import "github.com/ministryofjustice/opg-data-lpa-store/internal/shared"
+
+type AttorneyOptOut struct {
+	AttorneyUID string
+}
+
+func (c AttorneyOptOut) Apply(lpa *shared.Lpa) []shared.FieldError {
+	attorney, ok := lpa.GetAttorney(c.AttorneyUID)
+	if !ok {
+		return []shared.FieldError{{Source: "/type", Detail: "attorney not found"}}
+	}
+
+	if attorney.SignedAt != nil && !attorney.SignedAt.IsZero() {
+		return []shared.FieldError{{Source: "/type", Detail: "attorney cannot opt out after signing"}}
+	}
+
+	// TODO Check if status correct
+	attorney.Status = shared.AttorneyStatusRemoved
+	lpa.PutAttorney(attorney)
+
+	switch len(lpa.Attorneys) {
+	case 1, 2:
```

Comment:
Should this be checking that there are 1 or 2 _active_ attorneys? In case they've already been removed.

Reply:
oh yeah, and will also need to take into account trust corporations and replacements

Reply:
Good call on the attorney status, I'll update.

I was planning on handling replacements and trust corps in a separate ticket as the logic gets a bit spicy then but will double check if there's a ticket in the backlog and add one if not.